In [1]:
#Import Package
import pandas as pd
import numpy as np

# Load the Data
#### Here, we load the data in the variable 'data' and display it

In [2]:
data = pd.read_csv('Dataset.csv')

In [3]:
data

Gender    Height      Weight  Index
0       Male  1.875714  109.721205      4
1       Male  1.747060   73.622879      1
2       Male  1.882397   96.497744      2
3       Male  1.821967   99.809704      3
4       Male  1.774998   93.598806      3
...      ...       ...         ...    ...
9995  Female  1.680785   62.041283      1
9996  Female  1.703506   77.504471      4
9997  Female  1.622247   58.275494      2
9998  Female  1.753470   74.322315      3
9999  Female  1.573384   51.550427      1

[10000 rows x 4 columns]

# Describe the Data in Multiple Ways

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Gender  10000 non-null  object 
 1   Height  10000 non-null  float64
 2   Weight  10000 non-null  float64
 3   Index   10000 non-null  int64  
dtypes: float64(2), int64(1), object(1)
memory usage: 312.6+ KB


#### By using info(), we saw the the type of the data and the rows and columns it consists. It also shows us the total  memory space of the dataset.

In [10]:
data.describe()

Height        Weight         Index
count  10000.000000  10000.000000  10000.000000
mean       1.685736     73.228201      2.222200
std        0.097727     14.564160      0.865507
min        1.378284     29.347519      0.000000
25%        1.613043     61.606105      2.000000
50%        1.684479     73.125041      2.000000
75%        1.757026     84.898769      3.000000
max        2.006568    122.465413      4.000000

#### describe() shows us the descriptive statistics of the dataset. 

In [4]:
data.Gender

0         Male
1         Male
2         Male
3         Male
4         Male
         ...  
9995    Female
9996    Female
9997    Female
9998    Female
9999    Female
Name: Gender, Length: 10000, dtype: object

#### This shows the information of only 'Gender' attribute, and describes it.

In [5]:
data.Weight

0       109.721205
1        73.622879
2        96.497744
3        99.809704
4        93.598806
           ...    
9995     62.041283
9996     77.504471
9997     58.275494
9998     74.322315
9999     51.550427
Name: Weight, Length: 10000, dtype: float64

#### This shows the information of only 'Weight' attribute, and descibes it.

In [7]:
data.Height

0       1.875714
1       1.747060
2       1.882397
3       1.821967
4       1.774998
          ...   
9995    1.680785
9996    1.703506
9997    1.622247
9998    1.753470
9999    1.573384
Name: Height, Length: 10000, dtype: float64

#### This shows the information of only 'Height' attribute, and descibes it.

In [8]:
data.Index

0       4
1       1
2       2
3       3
4       3
       ..
9995    1
9996    4
9997    2
9998    3
9999    1
Name: Index, Length: 10000, dtype: int64

#### This shows the information of only 'Index' attribute, and descibes it.

In [12]:
data = pd.read_csv('Dataset.csv')
data.shape

(10000, 4)

#### This shows the shape of the data, which has 1000 rows and 4 columns.

In [22]:
data = pd.read_csv('Dataset.csv')
data.columns

Index(['Gender', 'Height', 'Weight', 'Index'], dtype='object')

#### It shows the name of the attributes of various columns in the data and its type.

In [23]:
data = pd.read_csv('Dataset.csv')
data.size

40000

#### It shows the total size of the dataset, which is 1000 rows and 4 columns.

In [24]:
data = pd.read_csv('Dataset.csv')
data.dtypes

Gender     object
Height    float64
Weight    float64
Index       int64
dtype: object

#### It shows the data type of each attribute.

In [25]:
data = pd.read_csv('Dataset.csv')
data.count

<bound method DataFrame.count of       Gender    Height      Weight  Index
0       Male  1.875714  109.721205      4
1       Male  1.747060   73.622879      1
2       Male  1.882397   96.497744      2
3       Male  1.821967   99.809704      3
4       Male  1.774998   93.598806      3
...      ...       ...         ...    ...
9995  Female  1.680785   62.041283      1
9996  Female  1.703506   77.504471      4
9997  Female  1.622247   58.275494      2
9998  Female  1.753470   74.322315      3
9999  Female  1.573384   51.550427      1

[10000 rows x 4 columns]>

####  count() is used to count the no. of non-NaN observations

# Remove insignificant columns

In [26]:
data = pd.read_csv('Dataset.csv')
data.isnull()

Gender  Height  Weight  Index
0      False   False   False  False
1      False   False   False  False
2      False   False   False  False
3      False   False   False  False
4      False   False   False  False
...      ...     ...     ...    ...
9995   False   False   False  False
9996   False   False   False  False
9997   False   False   False  False
9998   False   False   False  False
9999   False   False   False  False

[10000 rows x 4 columns]

#### We see that there is no insignificant value present in the dataset.

# Check Missing value for each columns

In [27]:
data = pd.read_csv('Dataset.csv')
data.isnull().sum()

Gender    0
Height    0
Weight    0
Index     0
dtype: int64

#### This tells us the total sum of NaN values presnt in each column, which in this case is already 0.

In [28]:
data.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
9995    False
9996    False
9997    False
9998    False
9999    False
Length: 10000, dtype: bool

#### There are no duplicte values in the data.

# Remove rows with missing values

In [29]:
data.isnull()

Gender  Height  Weight  Index
0      False   False   False  False
1      False   False   False  False
2      False   False   False  False
3      False   False   False  False
4      False   False   False  False
...      ...     ...     ...    ...
9995   False   False   False  False
9996   False   False   False  False
9997   False   False   False  False
9998   False   False   False  False
9999   False   False   False  False

[10000 rows x 4 columns]

#### False indicates that there are no NaN values present in the dataset.

# Formatting your Data – making data types compaitable with other data types

In [34]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Gender  10000 non-null  object 
 1   Height  10000 non-null  float64
 2   Weight  10000 non-null  float64
 3   Index   10000 non-null  int64  
dtypes: float64(2), int64(1), object(1)
memory usage: 312.6+ KB


#### Our datatypes are compatible with each other, hence there is no need to format it.

# Summary
#### Our dataset has 4 attributes:
>Gender, Height, Weight and Index
#### The datatype of gender is object.
#### The datatype of height is float64.
#### The datatype of weight is float64.
#### The datatype of index is int64.
#### There are no NaN values in any row or column.
#### The dataset has 10000 rows and 4 columns. Which means it has a total dataset of 9999 people.